hal/halx86/smp/smp.c

```diff
@@ -10,23 +10,48 @@
 
 #include <hal.h>
 #include <smp.h>
+#include "smpp.h"
 #define NDEBUG
 #include <debug.h>
 
-/* FUNCTIONS *****************************************************************/
+/* GLOBALS *******************************************************************/
 
-VOID
-NTAPI
-HalRequestIpi(KAFFINITY TargetProcessors)
-{
-    HalpRequestIpi(TargetProcessors);
-}
+extern PHYSICAL_ADDRESS HalpLowStubPhysicalAddress;
+extern PVOID HalpLowStub;
+
+/* TODO: MaxAPCount should be assigned by a Multi APIC table */
+ULONG MaxAPCount = 8;
+ULONG StartedProcessorCount = 1;
+
+/* FUNCTIONS *****************************************************************/
 
 BOOLEAN
 NTAPI
 HalStartNextProcessor(IN PLOADER_PARAMETER_BLOCK LoaderBlock,
                       IN PKPROCESSOR_STATE ProcessorState)
 {
-    //@Unimplemented
-    return FALSE;
+    if(MaxAPCount > StartedProcessorCount)
+    {
+        /* We only a create a new pagetable once */
+        if(StartedProcessorCount == 1)
+        {
+            //HalpInitializeAPPageTables(HalpLowStub);
+        }
+        
+        /* Start an AP */
+        HalpInitializeAPStub(HalpLowStub);
```

Comment:
Maybe call this only when `StartedProcessorCount == 1` ? Or on HAL entry point ?

Reply:
At the stub expands into the next PR, we overwrite the stub per AP as we use different values per application processor.

Reply:
👍